packages/nc-gui-v2/components/cell/Json.vue

```diff
@@ -0,0 +1,132 @@
+<script setup lang="ts">
+import { Modal as AModal } from 'ant-design-vue'
+import Editor from '~/components/monaco/Editor.vue'
+import FullScreenIcon from '~icons/cil/fullscreen'
+import FullScreenExitIcon from '~icons/cil/fullscreen-exit'
+import { inject, onMounted } from '#imports'
+import { EditModeInj } from '~/context'
+
+interface Props {
+  modelValue: string | Record<string, any> | undefined
+}
+
+interface Emits {
+  (event: 'update:modelValue', model: string): void
+}
+
+const props = defineProps<Props>()
+
+const emits = defineEmits<Emits>()
+
+const editEnabled = inject(EditModeInj, ref(false))
+
+let vModel = $(useVModel(props, 'modelValue', emits))
+
+let localValueState = $ref<string | undefined>(undefined)
+let localValue = $(
+  computed<string | undefined>({
+    get: () => localValueState,
+    set: (val: undefined | string | Record<string, any>) => {
+      localValueState = typeof val === 'object' ? JSON.stringify(val, null, 2) : val
+    },
+  }),
+)
+
+let error = $ref<string | undefined>(undefined)
+let isExpanded = $ref(false)
+
+const clear = () => {
+  error = undefined
+  isExpanded = false
+  editEnabled.value = false
+
+  localValue = vModel
+}
+
+const formatJson = (json: string) => {
+  try {
+    return JSON.stringify(JSON.parse(json), null, 2)
+  } catch (e) {
+    return json
+  }
+}
+
+const onSave = () => {
+  isExpanded = false
+  editEnabled.value = false
+  localValue = localValue ? formatJson(localValue) : localValue
+  vModel = localValue
+}
+
+onMounted(() => {
+  localValue = vModel
+})
+
+watch(
+  () => vModel,
+  (val) => {
+    localValue = val
+  },
+)
+
+watch(
+  () => localValue,
+  (val) => {
+    try {
+      JSON.parse(val)
+      error = undefined
+    } catch (e: any) {
+      error = e
+    }
+  },
+)
+
+watch(
+  () => editEnabled,
```

Comment:
instead of `() => editEnabled` you can also just pass `editEnabled` to the watcher.
Watchers accept refs/computeds as a watch source, doesn't have to be a function getter.